src/Tizen.NUI/src/public/BaseComponents/TextGeometry.cs

```diff
@@ -0,0 +1,182 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+using System;
+using System.ComponentModel;
+using Tizen.NUI.Text;
+using System.Collections.Generic;
+
+namespace Tizen.NUI.BaseComponents
+{
+    /// <summary>
+    /// </summary>
+    // This will be public opened after ACR done. (Before ACR, need to be hidden as Inhouse API)
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public static partial class TextGeometry
+    {
+        private static void ValidateRange(int start, int end)
+        {
+            if (start < 0)
+                throw new global::System.ArgumentOutOfRangeException(nameof(start), "Value is less than zero");
+            if (end < 0)
+                throw new global::System.ArgumentOutOfRangeException(nameof(end), "Value is less than zero");
+        }
+
+        private static List<Size2D> GetSizeListFromNativeVector(System.IntPtr ptr)
+        {
+            VectorVector2 sizeVector = new VectorVector2 (ptr, true);
+            int count = sizeVector.Size();
+            List<Size2D> list = new List<Size2D>();
+
+            for(int i = 0; i < count; i++)
+                list.Add(sizeVector.ValueOfIndex( (uint)i ));
+
+            return list;
+        }
+
+        private static List<Position2D> GetPositionListFromNativeVector(System.IntPtr ptr)
+        {
+            VectorVector2 positionVector = new VectorVector2 (ptr, true);
+            int count = positionVector.Size();
+            List<Position2D> list = new List<Position2D>();
+
+            for(int i = 0; i < count; i++)
+                list.Add(positionVector.ValueOfIndex( (uint)i ));
+
+            return list;
+        }
+
+        private static void CheckSWIGPendingException()
+        {
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+        }
+
+        /// <summary>
+        /// Get the rendered size of the text between start and end (included). <br />
+        /// if the requested text is at multilines, multiple sizes will be returned for each text located in a separate line. <br />
+        /// if a line contains characters with different directions, multiple sizes will be returned for each block of contiguous characters with the same direction. <br />
+        /// </summary>
+        /// <param name="textEditor">The TextEditor control containing the text.</param>
+        /// <param name="start">The start index of the text to get the size for</param>
+        /// <param name="end">The end index of the text to get the size for.</param>
+        // This will be public opened after ACR done. (Before ACR, need to be hidden as Inhouse API)
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static List<Size2D> GetTextSize(TextEditor textEditor, int start, int end)
```

Comment:
I don't mind change it to be extension, but it was requested to be in geometry class (to be like CopyToClipboard in TextUtils class).
also if we should use extension methods, then why not just add them directly inside TextEditor Class ?

Reply:
I think it's a good idea.
If we add directly to TextEditor, there is a problem that duplicate code is created in all Text Components.

I think it would be good to consider providing the Copy, Cut, Paste API as same way (extension method).

Reply:
I don't think we will have a duplicated code, we can move the current methods in TextGeometry class to : 
1- create internal class for example TextGeometryHelper that contains :
```
   void ValidateRange(...)
   List<Size2D> GetSizeListFromNativeVector(....)
   List<Position2D> GetPositionListFromNativeVector(...)
```
2- move GetTextSize/GetTextPosition methods to TextEditor/TextLabel/TextField

what do you think about it ? if this cannot be done we can return to the extension solution.

Reply:
It doesn't seem appropriate to provide each TextLabel, TextField, and TextEditor with a property or method that performs the same role.
I think it's a design problem that doesn't have a superclass like the Text class.

It is difficult to solve right away, but I think that providing the TextGeomtry class is better in terms of extension and maintenance.
I thought that the extension method could be provided in terms of convenience.

So, moving the GetTextSize/GetTextPosition methods to TextEditor/TextLabel/TextField doesn't seem appropriate.

Anyway, I don't think this should be decided right away.
It's worth discussing further later how to provide extension methods with copy, cut and paste.

Reply:
yes I agree it is a design problem.
If I found any ideas about it, I will share it with you.